core/scripts/cli/voting/filterRequestsByRound.js

```diff
@@ -0,0 +1,41 @@
+const { computeTopicHash } = require("../../../../common/EncryptionHelper");
+const { VotePhasesEnum } = require("../../../../common/Enums");
+
+module.exports = async (pendingRequests, account, roundId, roundPhase, votingContract) => {
+  let filteredRequests = [];
+  if (pendingRequests.length > 0) {
+    // Sort requests by timestamp requested
+    const chronologicalPriceRequests = pendingRequests.sort((a, b) => {
+      return parseInt(a.time) - parseInt(b.time);
+    });
+
+    // Depending on the round phase, determine which requests to display
+    if (roundPhase.toString() === VotePhasesEnum.COMMIT) {
+      // Display all requests during commit phase even if
+      // user has already committed a vote, for they
+      // might want to change it
+      for (let i = 0; i < chronologicalPriceRequests.length; i++) {
+        const request = chronologicalPriceRequests[i];
+        filteredRequests.push({
+          identifier: request.identifier,
+          time: request.time
+        });
+      }
+    } else {
+      // Only display committed votes during the reveal phase (i.e.
+      // if an encrypted message exists for the identifier-timestamp)
+      for (let i = 0; i < chronologicalPriceRequests.length; i++) {
```

Comment:
nit: this one is more optional (may be cleaner the way you have it), but wanted to point out a slightly different way of doing this:
```
filteredRequests = chronologicalPriceRequests.filter(request => {
  const topicHash = computeTopicHash(request, roundId);
  const encryptedCommit = await votingContract.getMessage(account, topicHash, { from: account });
  return encryptedCommit;
});
```

Reply:
yeah this is more the more functional way to do it and saves a couple lines, I implemented it